src/sdsc/__init__.py

```diff
@@ -607,6 +607,18 @@ def preparepatterns(patterngroupxpath, useonepattern):
     return [patternsofpatterngroup, patternforonepattern]
 
 
+def contextpatternlocations(locationxpath, factors=[1], fuzzymode=False):
```

Comment:
"locationxpath" is somewhat of a bad variable name, because there is no XPath at this point. Just "location"?